integration/Substra/plot_substrafl_flamby.py

```diff
@@ -0,0 +1,427 @@
+    you can run the command: `pip install -r requirements.txt` to install them.
+
+  - **Substra** and **Substrafl** should already be installed. If not, follow the instructions described here:
+    :ref:`get_started/installation:Installation`
+
+**Objective:**
+
+This example will compare a federated training on all 6 centers.
+
+This example shows how to interface Substra with Flamby. The modifications to use this example on other dataset are minimal
+and have to be made on the openers files, the metric file and the main script.
+"""
+
+# %%
+# Client and data preparation
+# ***************************
+#
+# Imports
+# =======
+import os
+import pathlib
+import copy
+import zipfile
+
+# %%
+# Creating the Substra Client
+# ===========================
+#
+# We work with six different organizations, defined by their IDs. Both organizations provide a dataset. One of them
+# will also provide the algorithm and will register the machine learning tasks.
+#
+# Once these variables defined, we can create our Substra :ref:`documentation/references/sdk:Client`.
+#
+# This example runs in local mode, simulating a **federated learning** experiment.
+
+from substra import Client
+
+N_CLIENTS = 6
+
+# Create the substra clients
+clients = [Client(debug=True) for _ in range(N_CLIENTS)]
+clients = {client.organization_info().organization_id: client for client in clients}
+
+# Store their IDs
+ORGS_ID = list(clients.keys())
+
+# The org id on which your computation tasks are registered
+ALGO_ORG_ID = ORGS_ID[0]
+
+# Choose the subprocess mode to locally simulate the FL process
+DEBUG_SPAWNER = "subprocess"
+os.environ["DEBUG_SPAWNER"] = DEBUG_SPAWNER
+
+
+assets_directory = pathlib.Path.cwd() / "assets"
+
+
+# %%
+# Registering assets
+# ******************
+#
+# Substra and Substrafl imports
+# ==============================
+
+from substra.sdk.schemas import (
+    DatasetSpec,
+    AlgoInputSpec,
+    AlgoOutputSpec,
+    AssetKind,
+    Permissions,
+    DataSampleSpec,
+    AlgoCategory,
+    AlgoSpec,
+)
+from substrafl.nodes import TestDataNode, TrainDataNode
+
+# %%
+# Permissions
+# ===========
+#
+# As data can not be seen once it is registered on the platform, we set :ref:`documentation/concepts:Permissions` for
+# each :ref:`documentation/concepts:Assets` define their access rights to the different data.
+#
+# The metadata are visible by all the users of a :term:`Channel`.
+
+permissions = Permissions(public=False, authorized_ids=ORGS_ID)
+
+# %%
+# Adding Metrics
+# ==============
+#
+# A metric corresponds to an algorithm used to compute the score of predictions on a
+# **datasample**.
+# Concretely, a metric corresponds to an archive *(tar or zip file)*, automatically build
+# from:
+#
+# - a **Python scripts** that implement the metric computation
+# - a `Dockerfile <https://docs.docker.com/engine/reference/builder/>`__ to specify the required dependencies of the
+#   **Python scripts**
+
+
+inputs_metrics = [
```

Comment:
Could you explain what's going on here ?

Reply:
This would ask a lot of effort and and might add confusions to explain everything that is happening here... I will use a Joker for now and discuss it with the squad.

Reply:
I am still overly confused by this code, but this should be dealt with on the substrafl's side.